packages/ra-core/src/util/FieldTitle.tsx

```diff
@@ -33,7 +36,7 @@ export const FieldTitle = (props: FieldTitleProps) => {
             {translateLabel({
                 label,
                 resource,
-                source,
+                source: finalSource,
```

Comment:
No, otherwise the inputs in an ArrayInput will all have an (untranslatable) name.

Reply:
True, but we probably don't want the input name alone. What happen if you have two fields with the same source, one at the record root and another in the ArrayInput?

Reply:
Actually, it seems we had this before: `resources.RESOURCE.fields.ARRAY_INPUT_SOURCE.INPUT_SOURCE`

Reply:
Well this change allows the FieldTitle to behave as before. That can be verified with the `calls its children with updated scopedFormData when inside an ArrayInput` test of the `FormDataConsumer`